src/lib/addons/feature-event-formatter-md.ts

```diff
@@ -82,307 +21,39 @@ export enum LinkStyle {
-        path: '/admin/message-banners',
-    },
-    [BANNER_DELETED]: {
-        label: 'Banner deleted',
-        action: '*{{user}}* deleted banner *{{event.preData.message}}*',
-        path: '/admin/message-banners',
-    },
-    [BANNER_UPDATED]: {
-        label: 'Banner updated',
-        action: '*{{user}}* updated banner *{{event.preData.message}}*',
-        path: '/admin/message-banners',
-    },
-    [PROJECT_CREATED]: {
-        label: 'Project created',
-        action: '*{{user}}* created project *{{project}}*',
-        path: '/projects',
-    },
-    [PROJECT_DELETED]: {
-        label: 'Project deleted',
-        action: '*{{user}}* deleted project *{{event.project}}*',
-        path: '/projects',
-    },
-    [SEGMENT_CREATED]: {
-        label: 'Segment created',
-        action: '*{{user}}* created segment *{{event.data.name}}*',
-        path: '/segments',
-    },
-    [SEGMENT_DELETED]: {
-        label: 'Segment deleted',
-        action: '*{{user}}* deleted segment *{{event.preData.name}}*',
-        path: '/segments',
-    },
-    [SEGMENT_UPDATED]: {
-        label: 'Segment updated',
-        action: '*{{user}}* updated segment *{{event.preData.name}}*',
-        path: '/segments',
-    },
-    [SERVICE_ACCOUNT_CREATED]: {
-        label: 'Service account created',
-        action: '*{{user}}* created service account *{{event.data.name}}*',
-        path: '/admin/service-accounts',
-    },
-    [SERVICE_ACCOUNT_DELETED]: {
-        label: 'Service account deleted',
-        action: '*{{user}}* deleted service account *{{event.preData.name}}*',
-        path: '/admin/service-accounts',
-    },
-    [SERVICE_ACCOUNT_UPDATED]: {
-        label: 'Service account updated',
-        action: '*{{user}}* updated service account *{{event.preData.name}}*',
-        path: '/admin/service-accounts',
-    },
-    [USER_CREATED]: {
-        label: 'User created',
-        action: '*{{user}}* created user *{{event.data.name}}*',
-        path: '/admin/users',
-    },
-    [USER_DELETED]: {
-        label: 'User deleted',
-        action: '*{{user}}* deleted user *{{event.preData.name}}*',
-        path: '/admin/users',
-    },
-    [USER_UPDATED]: {
-        label: 'User updated',
-        action: '*{{user}}* updated user *{{event.preData.name}}*',
-        path: '/admin/users',
-    },
-};
+type FormatStyle = 'simple' | 'markdown';
+
+interface IFeatureEventFormatterMdArgs {
+    unleashUrl: string;
+    linkStyle?: LinkStyle;
+    formatStyle?: FormatStyle;
+}
 
 export class FeatureEventFormatterMd implements FeatureEventFormatter {
     private readonly unleashUrl: string;
 
     private readonly linkStyle: LinkStyle;
 
-    constructor(unleashUrl: string, linkStyle: LinkStyle = LinkStyle.MD) {
+    private readonly formatStyle: FormatStyle;
+
+    constructor({
+        unleashUrl,
+        linkStyle = LinkStyle.MD,
+        formatStyle = 'simple',
+    }: IFeatureEventFormatterMdArgs) {
```

Comment:
Named parameters improve code readability and make function calls more flexible by allowing arguments to be passed in any order.